src/ImageSharp/Formats/WebP/LosslessUtils.cs

```diff
@@ -0,0 +1,656 @@
+    /// Utility functions for the lossless decoder.
+    /// </summary>
+    internal static class LosslessUtils
+    {
+        /// <summary>
+        /// Add green to blue and red channels (i.e. perform the inverse transform of 'subtract green').
+        /// </summary>
+        /// <param name="pixelData">The pixel data to apply the transformation.</param>
+        public static void AddGreenToBlueAndRed(Span<uint> pixelData)
+        {
+            for (int i = 0; i < pixelData.Length; i++)
+            {
+                uint argb = pixelData[i];
+                uint green = (argb >> 8) & 0xff;
+                uint redBlue = argb & 0x00ff00ffu;
+                redBlue += (green << 16) | green;
+                redBlue &= 0x00ff00ffu;
+                pixelData[i] = (argb & 0xff00ff00u) | redBlue;
+            }
+        }
+
+        /// <summary>
+        /// If there are not many unique pixel values, it may be more efficient to create a color index array and replace the pixel values by the array's indices.
+        /// This will reverse the color index transform.
+        /// </summary>
+        /// <param name="transform">The transform data contains color table size and the entries in the color table.</param>
+        /// <param name="pixelData">The pixel data to apply the reverse transform on.</param>
+        public static void ColorIndexInverseTransform(Vp8LTransform transform, Span<uint> pixelData)
+        {
+            int bitsPerPixel = 8 >> transform.Bits;
+            int width = transform.XSize;
+            int height = transform.YSize;
+            Span<uint> colorMap = transform.Data.GetSpan();
+            int decodedPixels = 0;
+            if (bitsPerPixel < 8)
+            {
+                int pixelsPerByte = 1 << transform.Bits;
+                int countMask = pixelsPerByte - 1;
+                int bitMask = (1 << bitsPerPixel) - 1;
+
+                var decodedPixelData = new uint[width * height];
+                int pixelDataPos = 0;
+                for (int y = 0; y < height; ++y)
+                {
+                    uint packedPixels = 0;
+                    for (int x = 0; x < width; ++x)
+                    {
+                        // We need to load fresh 'packed_pixels' once every
+                        // 'pixelsPerByte' increments of x. Fortunately, pixelsPerByte
+                        // is a power of 2, so can just use a mask for that, instead of
+                        // decrementing a counter.
+                        if ((x & countMask) is 0)
+                        {
+                            packedPixels = GetArgbIndex(pixelData[pixelDataPos++]);
+                        }
+
+                        decodedPixelData[decodedPixels++] = colorMap[(int)(packedPixels & bitMask)];
+                        packedPixels >>= bitsPerPixel;
+                    }
+                }
+
+                decodedPixelData.AsSpan().CopyTo(pixelData);
+
+                return;
```

Comment:
I really don't like early returns. Makes code harder to read, but if I recall right, I'm alone or part of a minority with this opinion for ImageSharp. :(
(Arguments against this:
- not faster than else, as the compiler generates this return itself.
- this is semantically an if-then-else construct, but it looks like an if-then - and always do something else.
  The return changes the semantics of the other language construct, and a return in a block is easily hidden when the block is collapsed in an IDE or similar.